src/storage-preview/azext_storage_preview/_client_factory.py

```diff
@@ -136,13 +137,35 @@ def cf_mgmt_file_services(cli_ctx, _):
 
 
 def get_account_url(cli_ctx, account_name, service):
-    from knack.util import CLIError
+    from azure.cli.core.azclierror import RequiredArgumentMissingError
     if account_name is None:
-        raise CLIError("Please provide storage account name or connection string.")
+        raise RequiredArgumentMissingError("Please provide storage account name or connection string.")
     storage_endpoint = cli_ctx.cloud.suffixes.storage_endpoint
     return "https://{}.{}.{}".format(account_name, service, storage_endpoint)
 
 
+def cf_queue_service(cli_ctx, kwargs):
+    t_queue_service = get_sdk(cli_ctx, CUSTOM_DATA_STORAGE_QUEUE, '_queue_service_client#QueueServiceClient')
+    connection_string = kwargs.pop('connection_string', None)
+    account_name = kwargs.pop('account_name', None)
+    account_key = kwargs.pop('account_key', None)
+    token_credential = kwargs.pop('token_credential', None)
+    sas_token = kwargs.pop('sas_token', None)
+    if connection_string:
+        return t_queue_service.from_connection_string(conn_str=connection_string)
+
+    account_url = get_account_url(cli_ctx, account_name=account_name, service='queue')
+    credential = account_key or sas_token or token_credential
+
+    if account_url and credential:
```

Comment:
Could you apply these changes here https://github.com/Azure/azure-cli/blob/1c8caadaf218fa25c924d55884a0a22a3eb0fdfb/src/azure-cli/azure/cli/command_modules/storage/_client_factory.py#L303-L306
because as you can see the SDK could accept credentials=None

https://github.com/Azure/azure-cli-extensions/blob/54c047982d1059b8daf516e627b3c4f1296023fc/src/storage-preview/azext_storage_preview/vendored_sdks/azure_storage_queue/v2018_03_28/_queue_client.py#L76

Reply:
done